server/src/main/java/com/cloud/network/Ipv6AddressManagerImpl.java

```diff
@@ -195,13 +195,23 @@ protected boolean isIp6Taken(Network network, String requestedIpv6) {
      * address information.
      */
     @Override
-    public void setNicIp6Address(final NicProfile nic, final DataCenter dc, final Network network) {
+    public void setNicIp6Address(final NicProfile nic, final DataCenter dc, final Network network) throws InsufficientAddressCapacityException {
         if (network.getIp6Gateway() != null) {
             if (nic.getIPv6Address() == null) {
                 s_logger.debug("Found IPv6 CIDR " + network.getIp6Cidr() + " for Network " + network);
                 nic.setIPv6Cidr(network.getIp6Cidr());
                 nic.setIPv6Gateway(network.getIp6Gateway());
 
+                if (nic.getBroadcastType() == null) {
+                    nic.setBroadcastType(network.getBroadcastDomainType());
+                }
+                if (nic.getBroadCastUri() == null) {
+                    nic.setBroadcastUri(network.getBroadcastUri());
+                }
+                if (nic.getMacAddress() == null) {
+                    nic.setMacAddress(_networkModel.getNextAvailableMacAddressInNetwork(network.getId()));
+                }
+
```

Comment:
I sugest a method `setNicPropertiesFromNetwork(NicProfile nic, Network network)`

Reply:
done. thanks @sureshanaparti @DaanHoogland @GabrielBrascher for review.